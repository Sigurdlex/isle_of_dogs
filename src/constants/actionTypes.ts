export const FETCH_ALL_BREEDS = 'FETCH_ALL_BREEDS';
export const FETCH_SINGLE_BREED = 'FETCH_SINGLE_BREED';
export const SET_DISPLAYED_BREEDS = 'SET_DISPLAYED_BREEDS';
export const CHANGE_MY_BREEDS = 'CHANGE_MY_BREEDS';
export const SET_BREEDS_FILTER = 'SET_BREEDS_FILTER';
export const ENTER_SEARCH = 'ENTER_SEARCH';
export const SET_IS_FETCHING = 'SET_IS_FETCHING';
export const SET_IMAGES = 'SET_IMAGES';
