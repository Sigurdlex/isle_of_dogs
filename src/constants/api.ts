export const API_URL = 'https://dog.ceo/api/';
export const ALL_BREEDS_ENDP = 'breeds/list/all';
export const SINGLE_BREED_ENDP = (breed: string) => `breed/${breed}/images`;
