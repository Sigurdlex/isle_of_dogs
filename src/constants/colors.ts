export const ROYALBLUE = '#3d3afd';
export const LIMEGREEN = '#10cd5b';
export const ORANGE = '#ffbd03';
export const LIGHTGRAY = '#d6d6d6';
