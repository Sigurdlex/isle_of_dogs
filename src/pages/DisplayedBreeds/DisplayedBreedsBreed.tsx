import React, { useCallback } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Dispatch } from 'redux';
import styled from 'styled-components';

import { changeMyBreeds } from 'actionCreators';
import { LIMEGREEN, ORANGE, ROYALBLUE } from 'constants/colors';
import { FilledButton } from 'components';

export interface DisplayedBreedsBreedProps {
  breed: string;
  mine: boolean;
  dispatch: Dispatch;
}

export const DisplayedBreedsBreed = (props: DisplayedBreedsBreedProps) => {
  const { breed, mine, dispatch } = props;
  const handleToggleClick = useCallback(() => dispatch(changeMyBreeds(breed)), [breed, dispatch]);
  return (
    <Container>
      <StyledLink to={`/breed/${breed}`}>{breed}</StyledLink>
      <FilledButton color={mine ? LIMEGREEN : ORANGE} handleClick={handleToggleClick}>
        {mine ? 'Remove from favourites' : 'Add to favourites'}
      </FilledButton>
    </Container>
  );
};

export default connect()(DisplayedBreedsBreed);

const Container = styled.div`
  padding: 0 3em;
  height: 3em;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 80%;
  margin-bottom: 1em;
  background: white;
  position: relative;
  z-index: 1;
  &::before,
  &::after {
    content: '';
    position: absolute;
    height: 100%;
    width: 5px;
    background: ${ROYALBLUE};
  }
  &::before {
    left: 1em;
  }
  &::after {
    right: 1em;
  }
  &:first-child {
    margin-top: 1em;
  }
  & > button {
    width: 170px;
  }
`;

const StyledLink = styled(Link)`
  color: ${ROYALBLUE};
`;
