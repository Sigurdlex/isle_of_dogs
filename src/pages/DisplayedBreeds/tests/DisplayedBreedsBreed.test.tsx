import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Dispatch } from 'redux';

import { ORANGE, LIMEGREEN } from 'constants/colors';
import { DisplayedBreedsBreed } from '../DisplayedBreedsBreed';

const breed = 'cerberus';
const dispatch: Dispatch = jest.fn() as Dispatch;

afterEach(() => {
  cleanup();
});

test('<DisplayBreedsBreed /> render with emtpy initial redux state, and breed isn\'t mine', () => {
  const { getByText, container: { firstChild: container } } = render(
    <MemoryRouter>
      <DisplayedBreedsBreed breed={breed} mine={false} dispatch={dispatch} />
    </MemoryRouter>
  );
  const button: HTMLButtonElement = getByText('Add to favourites') as HTMLButtonElement;
  expect(button).toHaveStyleRule('background', ORANGE);
  expect(container).toMatchSnapshot();
});

test('<DisplayBreedsBreed /> render with emtpy initial redux state, and breed is mine', () => {
  const { getByText, container: { firstChild: container } } = render(
    <MemoryRouter>
      <DisplayedBreedsBreed breed={breed} mine dispatch={dispatch} />
    </MemoryRouter>
  );
  const button: HTMLButtonElement = getByText('Remove from favourites') as HTMLButtonElement;
  expect(button).toHaveStyleRule('background', LIMEGREEN);
  expect(container).toMatchSnapshot();
});
