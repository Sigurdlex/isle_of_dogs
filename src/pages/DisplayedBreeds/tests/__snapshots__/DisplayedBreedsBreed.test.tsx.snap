// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayBreedsBreed /> render with emtpy initial redux state, and breed is mine 1`] = `
.c2 {
  color: white;
  font-weight: bold;
  border: 2px solid #10cd5b;
  background: #10cd5b;
  border-radius: 4px;
  height: 2em;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c2:hover {
  color: #10cd5b;
  background: transparent;
}

.c0 {
  padding: 0 3em;
  height: 3em;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.5);
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 80%;
  margin-bottom: 1em;
  background: white;
  position: relative;
  z-index: 1;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  height: 100%;
  width: 5px;
  background: #3d3afd;
}

.c0:before {
  left: 1em;
}

.c0:after {
  right: 1em;
}

.c0:first-child {
  margin-top: 1em;
}

.c0 > button {
  width: 170px;
}

.c1 {
  color: #3d3afd;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="/breed/cerberus"
  >
    cerberus
  </a>
  <button
    class="c2"
    color="#10cd5b"
  >
    Remove from favourites
  </button>
</div>
`;

exports[`<DisplayBreedsBreed /> render with emtpy initial redux state, and breed isn't mine 1`] = `
.c2 {
  color: white;
  font-weight: bold;
  border: 2px solid #ffbd03;
  background: #ffbd03;
  border-radius: 4px;
  height: 2em;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c2:hover {
  color: #ffbd03;
  background: transparent;
}

.c0 {
  padding: 0 3em;
  height: 3em;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.5);
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 80%;
  margin-bottom: 1em;
  background: white;
  position: relative;
  z-index: 1;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  height: 100%;
  width: 5px;
  background: #3d3afd;
}

.c0:before {
  left: 1em;
}

.c0:after {
  right: 1em;
}

.c0:first-child {
  margin-top: 1em;
}

.c0 > button {
  width: 170px;
}

.c1 {
  color: #3d3afd;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="/breed/cerberus"
  >
    cerberus
  </a>
  <button
    class="c2"
    color="#ffbd03"
  >
    Add to favourites
  </button>
</div>
`;
